tests/e2e/helpers/runners.py

```diff
@@ -197,8 +197,12 @@ def _run_once(
             finally:
                 chunk = _get_chunk(child)
                 output += chunk
-        if allow_nonzero_exitcode:
-            child.close()
+        if assert_exit_code:
+            if child.isalive():
+                # Short sleep to allow make command to complete
+                log_msg(f"Sleeping for {cmd}", logger=LOGGER.info)
+                time.sleep(1)
```

Comment:
what's the reason for this code? please provide either explanation/proof that we need it or let's drop it.

Reply:
and also explanation of this exact sleep delay.

Reply:
I thought the comment above these lines was clear, my bad.

My experiments indicate that after we find all required matching patterns it takes a tiny bit of time for `make` process to complete. If we call `close()` immediately, the process gets killed and hence returns a non-zero `signalstatus`.

As far as _exact sleep delay is concerned_, I'm afraid there's no such thing in asynchronous world. There's only good enough and not good enough delays. This one proved to be just fine in all my tests.

Reply:
what I know is that any sleep is a potential source of flakiness, this is my concern. As an alternative, can we use `child.close(force=False)`? Will it wait until the child is closed?

Reply:
> any sleep is a potential source of flakiness

I'm very well aware of that and agree with you. `close(force=False)` still sends `SIGTERM` to process which makes `signalstatus` non-zero, I tested that as well.

Reply:
should we use [`wait`](https://pexpect.readthedocs.io/en/stable/api/pexpect.html#pexpect.spawn.wait) instead?

Reply:
Already updated. Now dealing with NFS flakiness in an unrelated test

Reply:
All tests pass, yay!